server/app/services/program/ProgramServiceImpl.java

```diff
@@ -131,31 +131,32 @@ public CompletionStage<ProgramDefinition> getActiveProgramDefinitionAsync(String
   public CompletionStage<ImmutableList<ProgramDefinition>> syncQuestionsToProgramDefinitions(
       ImmutableList<ProgramDefinition> programDefinitions) {
 
-    /* TEMP BUG FIX
-     * Because some of the programs are not in the active version,
+    /* Because some of the programs are not in the active version,
      * and we need to sync the questions for each program to calculate
      * eligibility state, we must sync each program with a version it
      * is associated with. This diverges from previous behavior where
      * we did not need to sync the programs because the contents of their
      * questions was not needed in the index view.
      */
-    return CompletableFuture.completedFuture(
-        programDefinitions.stream()
-            .map(
-                programDef -> {
-                  Program p = programDef.toProgram();
-                  p.refresh();
-                  Version v = p.getVersions().stream().findAny().orElseThrow();
-                  try {
-                    return syncProgramDefinitionQuestions(
-                        programDef, questionService.getReadOnlyVersionedQuestionService(v));
-                    /* END TEMP BUG FIX */
-                  } catch (QuestionNotFoundException e) {
-                    throw new RuntimeException(
-                        String.format("Question not found for Program %s", programDef.id()), e);
-                  }
-                })
-            .collect(ImmutableList.toImmutableList()));
+    return CompletableFuture.supplyAsync(
+        () -> {
+          return programDefinitions.stream()
+              .map(
+                  programDef -> {
+                    Program p = programDef.toProgram();
+                    p.refresh();
+                    Version v = p.getVersions().stream().findAny().orElseThrow();
+                    try {
+                      return syncProgramDefinitionQuestions(
+                          programDef, questionService.getReadOnlyVersionedQuestionService(v));
+                    } catch (QuestionNotFoundException e) {
+                      throw new RuntimeException(
+                          String.format("Question not found for Program %s", programDef.id()), e);
+                    }
+                  })
+              .collect(ImmutableList.toImmutableList());
+        },
+        httpExecutionContext.current());
```

Comment:
**issue (blocking):** wrapping the contents of this method in `CompletableFuture.supplyAsync` and passing `httpExecutionContext.current` does not change its threading behavior from the previous implementation: an HTTP thread will still be blocked waiting for database IO.